bentoml/marshal/dispatcher.py

```diff
@@ -120,6 +120,17 @@ def __init__(
         self._queue = collections.deque()  # TODO(hrmthw): maxlen
         self._sema = shared_sema if shared_sema else NonBlockSema(1)
 
+    async def cleanup(self):
+        if self._controller is not None:
+            self._controller.cancel()
+            await asyncio.sleep(0)
+        try:
+            while True:
+                _, _, fut = self._queue.pop()
```

Comment:
Should we also add a condition to also reject all inbound calls?

Reply:
There will not be race conditions in an async APP.